Review prototype against registry ic-canister-sig-creation 1.3.1. Apache-2.0 upstream license remains applicable. Not selected by production.

diff --git a/Cargo.toml b/Cargo.toml
index 72bd90a..09d6a02 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,8 +50,8 @@ repository = "https://github.com/dfinity/ic-canister-sig-creation"
 name = "ic_canister_sig_creation"
 path = "src/lib.rs"
 
-[dependencies.candid]
-version = "0.10"
+[dependencies.ic_principal]
+version = "0.1"
 
 [dependencies.hex]
 version = "0.4"
@@ -62,8 +62,6 @@ version = "3.0"
 [dependencies.ic-representation-independent-hash]
 version = "3.0"
 
-[dependencies.ic0]
-version = "1.0.1"
 
 [dependencies.lazy_static]
 version = "1.4"
diff --git a/src/lib.rs b/src/lib.rs
index 588dfe3..81a8eda 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,4 +1,4 @@
-use candid::Principal;
+use ic_principal::Principal;
 use ic_certification::{Hash, HashTree};
 use ic_representation_independent_hash::{representation_independent_hash, Value};
 use lazy_static::lazy_static;
@@ -6,7 +6,7 @@ use serde::{Deserialize, Serialize};
 use serde_bytes::ByteBuf;
 use sha2::{Digest, Sha256};
 
-pub mod signature_map;
+
 
 pub const IC_ROOT_PK_DER_PREFIX: &[u8; 37] = b"\x30\x81\x82\x30\x1d\x06\x0d\x2b\x06\x01\x04\x01\x82\xdc\x7c\x05\x03\x01\x02\x01\x06\x0c\x2b\x06\x01\x04\x01\x82\xdc\x7c\x05\x03\x02\x01\x03\x61\x00";
 pub const IC_ROOT_PK_DER: &[u8; 133] = b"\x30\x81\x82\x30\x1d\x06\x0d\x2b\x06\x01\x04\x01\x82\xdc\x7c\x05\x03\x01\x02\x01\x06\x0c\x2b\x06\x01\x04\x01\x82\xdc\x7c\x05\x03\x02\x01\x03\x61\x00\x81\x4c\x0e\x6e\xc7\x1f\xab\x58\x3b\x08\xbd\x81\x37\x3c\x25\x5c\x3c\x37\x1b\x2e\x84\x86\x3c\x98\xa4\xf1\xe0\x8b\x74\x23\x5d\x14\xfb\x5d\x9c\x0c\xd5\x46\xd9\x68\x5f\x91\x3a\x0c\x0b\x2c\xc5\x34\x15\x83\xbf\x4b\x43\x92\xe4\x67\xdb\x96\xd6\x5b\x9b\xb4\xcb\x71\x71\x12\xf8\x47\x2e\x0d\x5a\x4d\x14\x50\x5f\xfd\x74\x84\xb0\x12\x91\x09\x1c\x5f\x87\xb9\x88\x83\x46\x3f\x98\x09\x1a\x0b\xaa\xae";
